### PR TITLE
fix(trace): Make dataset IDs unique by instance for TraceDataset

### LINKED ISSUE #2174: [ENHANCEMENT] TraceDataset.save() should have an optional path to save the file
As a user, I might want to save my trace outside of the working directory. It's also not clear what I'm supposed to do with the output ID. There should be `print` statements guiding the user as to how to use `load`

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #2174
Make dataset IDs unique by instance

### CODE REVIEW

File: src/phoenix/trace/trace_dataset.py [hunk 1]
```diff
@@ -98,15 +98,19 @@ class TraceDataset:
     """
 
     name: str
+    """
+    A human readable name for the dataset.
+    """
     dataframe: pd.DataFrame
     evaluations: List[Evaluations] = []
-    _id: UUID = uuid4()
+    _id: UUID
```
Comment (mikeldking) [inline src/phoenix/trace/trace_dataset.py:106]: this was making all trace datasets have the same ID I believe

File: src/phoenix/trace/trace_dataset.py [hunk 2]
```diff
@@ -98,15 +98,19 @@ class TraceDataset:
...
     _data_file_name: str = "data.parquet"
 
     def __init__(
         self,
         dataframe: DataFrame,
         name: Optional[str] = None,
+        id: Optional[Union[str, UUID]] = None,
```
Comment (axiomofjoy) [inline src/phoenix/trace/trace_dataset.py:113]: This doesn't seem like something we should expose on the `__init__`. Why not just initialize inside the `__init__`?

File: src/phoenix/trace/trace_dataset.py (hunk 1 above)
Comment (axiomofjoy) [reply src/phoenix/trace/trace_dataset.py:106]: Good catch.

**Review by axiomofjoy — APPROVED** [review]

File: src/phoenix/trace/trace_dataset.py (hunk 2 above)
Comment (mikeldking) [reply src/phoenix/trace/trace_dataset.py:113]: I sorta agree but I think if the user wants determinism we let them... It's a bit of a trade-off

**Pull request merged**